lib/thredded/per_notifier_pref.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+module Thredded
+  # in this class we dynamically set up an accessor for all the Thredded.notifiers (by their key)
+  # as we don't know in advance what the keys are,
+  # we try to not pollute the namespace
+  # nonetheless some keys would be bad, so we sanity check those on setup
+  class PerNotifierPref
```

Comment:
Yeah, I can see that's better. And that kind of structure will also be useful for queries  summary notifications (e.g. daily 'posts you haven't read today' -- in https://github.com/thredded/thredded/issues/344 that we just closed but I might just reopen and do after this). Just need to summon up the energy to refactor another time 😉   Will get to this after I finish #447